class.jetpack.php

```diff
@@ -4736,6 +4739,26 @@ function authenticate_jetpack( $user, $username, $password ) {
 		return new WP_User( $token_details['user_id'] );
 	}
 
+	// Authenticates requests from Jetpack server to WP API endpoints.
+	// Uses the existing XMLRPC oAuth implementation.
+	function wp_json_authenticate( $error ) {
```

Comment:
With this structure, even generic auth errors will win out, and Jetpack will never even get a chance to apply it's logic.

So I will proceed as such:
- If we see a `$_GET['token']` and `$_GET['signature']` we will assume it is Jetpack's job to provide auth logic, and all other auth will be ignored.
- On the Jetpack Server, we can also add an other parameter, eg `$_GET['jetpack_auth'] = true`, if we decide that we need to be more specific